core/types/client/opts.go

```diff
@@ -92,6 +98,14 @@ func WithChainID(chainID string) Option {
 	}
 }
 
+// WithTrustLocalChainID will trust the chain ID passed from the user, i.e. no check
+// against remote node's chain ID. This option is only be effective when ChainID is set.
+func WithTrustLocalChainID() Option {
```

Comment:
I feel I can actually just delete all `func Withxxx() Option` functions, since we only use `Options`

Reply:
This is a weird one.  I don't know why we have all the functional options funcs but the Options struct has settable exportable fields...  We should pick an approach.  I find functional options cumbersome, but I can live with it.  Whatever y'all want.

Reply:
This is easy pick, those are just relic code and we no longer used those `Option` type in sdk.
I'll delete those functions